tests/stress/um/stress_tests_um.cpp

```diff
@@ -0,0 +1,335 @@
+        REQUIRE(_xdp_program_info_provider != nullptr);
+
+        _test_control_info = get_jit_test_control_info();
+        if (_test_control_info.programs.size()) {
+
+            // Paranoia check - ensure that the program(s) we got back is/are indeed from our supported list.
+            for (const auto& program : _test_control_info.programs) {
+                if (std::find(_test_control_info.programs.begin(), _test_control_info.programs.end(), program) ==
+                    _test_control_info.programs.end()) {
+                    LOG_INFO("ERROR: Uexpected program: {}", program);
+                    REQUIRE(0);
+                }
+            }
+        } else {
+
+            // No programs specified on the command line, so use the preferred default.
+            _test_control_info.programs.push_back({"droppacket"});
+        }
+
+        LOG_INFO("One time initialization complete");
+        LOG_VERBOSE("jit test programs:");
+        for (const auto& program : _test_control_info.programs) {
+            LOG_VERBOSE("\t{}", program);
+        }
+        LOG_VERBOSE("jit test threads per program  : {}", _test_control_info.threads_count);
+        LOG_VERBOSE("jit test duration (in seconds): {}", _test_control_info.duration);
+        LOG_VERBOSE("jit test verbose output       : {}", _test_control_info.verbose_output);
+    });
+}
+
+TEST_CASE("load_attach_detach_unload_sequential_test", "[mt_stress_test]")
+{
+    um_test_init();
+
+    LOG_VERBOSE("Starting test: {}", Catch::getResultCapture().getCurrentTestName());
+
+    // Maintain an incrementing interface index counter to ensure that _each_ running thread gets a unique interface id.
+    // It is critical that a given interface index is associated with one and only one 'link' object in the entire set
+    // of running threads.
+    // The start value of 1 is for debugging purposes only. The user mode 'mock' framework does not validate this value
+    // in any fashion.
+    uint32_t if_index{1};
+
+    // This lambda creates the context for each thread, updates per-thread data therein and spawns the thread.  It also
+    // returns thread handle and thread context vectors for the created threads.
+    auto spawn_test_threads = [&](test_thread_function_t test_thread_function,
+                                  stress_test_thread_context& test_thread_context,
+                                  uint32_t test_thread_count) {
+        LOG_INFO("spawning stress test threads for {}...", test_thread_context.file_name.c_str());
+        std::vector<std::thread> tv(test_thread_count);
+        std::vector<stress_test_thread_context> ttc(test_thread_count);
+
+        for (uint32_t i = 0; i < test_thread_count; i++) {
+            LOG_VERBOSE("\t{}({}): Interface Index: {}", test_thread_context.file_name.c_str(), i, if_index);
+            ttc[i] = test_thread_context;
+            ttc[i].thread_index = if_index++;
```

Comment:
a "thread ID" and an "interface index" are different things.  Should the "thread_index" field be renamed to "if_index"?

Reply:
The ```thread_index``` member really is really meant to be used an index into the more-than-one arrays that the threads need to access and is used accordingly.  This value is guaranteed to be present in each thread context _and_ be unique.  This fit the bill for the if_index value requirements and I used it accordingly.

I can certainly add in a new ```if_index``` field in the context, or add a comment explaining the (overloaded) usage here.

Reply:
Added a comment explaining the overloaded usage.